lib/elixir/src/elixir_errors.erl

```diff
@@ -57,6 +61,125 @@ emit_diagnostic(Severity, Position, File, Message, Stacktrace) ->
 
   ok.
 
+% --------- Fancy Diagnostics
+
+fancy_exception(File, LineNumber, Column, Description, Snippet) ->
+  #{content := Content, offset := Offset} = Snippet,
+  LineDigits = get_line_number_digits(LineNumber),
+  Spacing = n_spaces(LineDigits + 1),
+  {FormattedSnippet, ColumnsTrimmed} = format_line(Content),
+
+  Formatted = io_lib:format(
+    " ~ts┌─ ~ts~ts\n"
+    " ~ts│\n"
+    " ~p │ ~ts\n"
+    " ~ts│ ~ts\n"
+    " ~ts│\n"
+    " ~ts~ts",
+    [
+     Spacing, prefix(error), file_format({LineNumber, Column}, File),
+     Spacing,
+     LineNumber, FormattedSnippet,
+     Spacing, highlight_below_line(Offset - ColumnsTrimmed, error),
+     Spacing,
+     Spacing, format_message(Description, LineDigits)
+    ]),
+
+  unicode:characters_to_binary(Formatted).
+
+fancy_exception(File, LineNumber, Column, Message) ->
+   Formatted = no_line_diagnostic({LineNumber, Column}, File, Message, error),
+   % Left pad so we stay aligned with "** (Exception)" banner
+   Padded = ["   ", string:replace(Formatted, "\n", "\n   ")],
+   unicode:characters_to_binary(Padded).
+
+no_line_diagnostic(Position, File, Message, Severity) -> 
+  io_lib:format(
+    " ┌─ ~ts~ts\n"
+    " ~ts",
+    [
+     prefix(Severity), file_format(Position, File),
+     Message
+    ]
+   ).
```

Comment:
You are missing `unicode:characters_to_binary` here. I think that would explain the rpc failure.